uhdm-tests/opentitan/Makefile.in

```diff
@@ -0,0 +1,123 @@
+OPEN_TITAN = ${curr_dir}/opentitan
+
+################################
+#### verilator - simulation ####
+################################
+OPEN_TITAN_BUILD = ${root_dir}/build/lowrisc_systems_top_earlgrey_verilator_0.1
+
+OPEN_TITAN_INCLUDE = \
+	-I$(OPEN_TITAN_BUILD)/src/lowrisc_prim_assert_0.1/rtl \
+	-I$(OPEN_TITAN_BUILD)/src/lowrisc_ibex_ibex_core_0.1/rtl \
+	-I$(OPEN_TITAN_BUILD)/src/lowrisc_dv_dv_macros_0 \
+	-I$(OPEN_TITAN_BUILD)/src/lowrisc_prim_util_memload_0/rtl
+
+# All Ibex Core files that are needed
+#TO_SURELOG='ibex_core.sv\|ibex_pkg.sv\|prim_clock_gating.sv\|prim_generic_clock_gating.sv\|ibex_if_stage.sv\|ibex_prefetch_buffer.sv\|ibex_compressed_decoder.sv\|ibex_fetch_fifo.sv\|ibex_id_stage.sv\|ibex_decoder.sv\|ibex_controller.sv\|ibex_ex_block.sv\|ibex_alu.sv\|ibex_multdiv_fast.sv\|ibex_load_store_unit.sv\|ibex_wb_stage.sv\|ibex_cs_registers.sv\|ibex_register_file_ff.sv\|ibex_csr.sv\|ibex_counter.sv\|ibex_icache.sv\|ibex_pmp.sv\|prim_secded_28_22_enc.sv\|prim_secded_72_64_enc.sv\|prim_ram_1p.sv\|prim_secded_28_22_dec.sv\|prim_secded_72_64_dec.sv\|prim_generic_ram_1p.sv'
+
+# Working set of Ibex Core files
+TO_SURELOG='ibex_pkg.sv\|prim_generic_clock_gating.sv\|ibex_prefetch_buffer.sv\|ibex_compressed_decoder.sv\|ibex_fetch_fifo.sv\|ibex_id_stage.sv\|ibex_decoder.sv\|ibex_controller.sv\|ibex_ex_block.sv\|ibex_alu.sv\|ibex_multdiv_fast.sv\|ibex_load_store_unit.sv\|ibex_wb_stage.sv\|ibex_csr.sv\|ibex_register_file_ff.sv\|ibex_pmp.sv\|prim_secded_28_22_enc.sv\|prim_secded_72_64_enc.sv\|prim_secded_28_22_dec.sv\|prim_secded_72_64_dec.sv'
+
+# In order to parse full Ibex Core it is needed to add:
+# \|ibex_core.sv\|prim_clock_gating.sv\|ibex_cs_registers.sv\|ibex_counter.sv\|ibex_if_stage.sv\|ibex_icache.sv\|prim_ram_1p.sv\|prim_generic_ram_1p.sv
+
+# BUILD BRAKERS:
+# prim_generic_ram_1p.sv
+# ibex_counter.sv
+# prim_clock_gating.sv (silent error: %Error: Exiting due to 1 error(s))
+
+# prim_ram_1p.sv (prim_generic_ram_1p.sv fault)
+# ibex_icache.sv (requires prim_ram_1p.sv)
+# ibex_cs_registers.sv (ibex_counter.sv fault)
+# ibex_if_stage.sv (requires ibex_icache.sv)
+# ibex_core.sv (requires ibex_if_stage.sv ibex_cs_registers.sv prim_clock_gating.sv)
+
+SKIP_SOURCES='rv_core_ibex_pkg.sv'
+
+# Need to readd this to .vc file due to double matching 'ibex_pkg.sv'
+READD='../src/lowrisc_ip_rv_core_ibex_pkg_0.1/rtl/rv_core_ibex_pkg.sv'
+
+# Top-level parameters in skipped modules that need to be known to Surelog
+SURELOG_PARAMETERS = +define+RVFI +define+VERILATOR_TEST_STATUS_ADDR=805306368 \
+			-PPMPEnable=1 \
+			-PPMPGranularity=0 \
+			-PPMPNumRegions=16 \
+			-PMHPMCounterNum=10 \
+			-PMHPMCounterWidth=32 \
+			-PRV32E=0 \
+			-PRV32M=3 \
+			-PRV32B=0 \
+			-PRegFile=0 \
+			-PBranchTargetALU=1 \
+			-PWritebackStage=1 \
+			-PICache=1 \
+			-PICacheECC=1 \
+			-PBranchPredictor=0 \
+			-PDbgTriggerEn=1 \
+			-PSecureIbex=0 \
+			-PDmHaltAddr=437323776 \
+			-PDmExceptionAddr=437323784 \
```

Comment:
Do we need all these parameters? According to Surelog documentation, it is only overriding top module parameters. You are not parsing ``ibex_core`` in Surelog, so Surelog automatically selects ``ibex_register_file_ff.sv`` as a top module and parameters in ``ibex_core`` should be correctly set in systemverilog frontend.

Reply:
I checked and it looks like we need those 3 parameters:
```
-PPMPNumRegions=16 \                                       
-PBranchTargetALU=1 \                                    
-PWritebackStage=1 \
```